x-pack/plugins/ml/public/application/jobs/new_job/job_from_lens/create_and_save_job.ts

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Embeddable } from '@kbn/lens-plugin/public';
+import type { DataViewsContract } from '@kbn/data-views-plugin/public';
+import type { TimefilterContract } from '@kbn/data-plugin/public';
+import type { IUiSettingsClient } from '@kbn/core/public';
+import { getJobsItemsFromEmbeddable } from './utils';
+import { createJob } from './create_job';
+import { MlApiServices } from '../../../services/ml_api_service';
+import type { ErrorType } from '../../../../../common/util/errors';
+
+interface State {
+  success: boolean;
+  error?: ErrorType;
+}
+
+interface CreateState {
+  jobCreated: State;
+  datafeedCreated: State;
+  jobOpened: State;
+  datafeedStarted: State;
+}
+
+export async function createAndSaveJob(
+  jobId: string,
+  bucketSpan: string,
+  embeddable: Embeddable,
+  startJob: boolean,
+  runInRealTime: boolean,
+  dataViewClient: DataViewsContract,
+  kibanaConfig: IUiSettingsClient,
+  timeFilter: TimefilterContract,
+  ml: MlApiServices,
+  layerIndex?: number
+): Promise<CreateState> {
+  const { query, filters, to, from, vis } = getJobsItemsFromEmbeddable(embeddable);
+  if (query === undefined || filters === undefined) {
+    throw new Error('Cannot create job, query and filters are undefined');
+  }
+
+  const { jobConfig, datafeedConfig, start, end } = await createJob(
```

Comment:
I am getting a duplicate `match_all` in the datafeed query this is being created:

```
    "query": {
      "bool": {
        "must": [
          {
            "match_all": {}
          },
          {
            "match_all": {}
          }
        ],
        "filter": [],
        "must_not": []
      }
    },
```

Reply:
Updated in https://github.com/elastic/kibana/pull/136421/commits/ae35f0de4bed8502f2c0f97cacee58d0675c6f29